jwst/pipeline/calwebb_spec2.py

```diff
@@ -237,8 +238,26 @@ def process_exposure_product(
                         raise RuntimeError('Cannot determine WCS.')
 
         # Steps whose order is the same for all types of input.
+
+        # Leakcal subtraction (imprint)  occurs before background subtraction on a per-exposure basis.
+        # If there is only one `imprint` member, this imprint exposure is subtracted from all the
+        # science and background exposures.  Otherwise, there will be as many `imprint` members as
+        # there are science plus background members. Which imprint to subtract from which science/background
+        # is determined by the matching values of the header keyword PATT_NUM, or
+        # `DataModel.meta.dither.position_number`.
+
+        # find the position number of all the leak cal observations
+        imprint_pos_no = self._imprint_pos_no(members_by_type)
```

Comment:
The function of identifying the imprints to use for the given input is entirely an internal function of the `imprint_step` itself. As such, this function should be moved from here to being defined in and being done by the `imprint_step` and removed from `calwebb_spec2` and removing the `imprint_pos_no` argument on the `imprint_step.process`.

This simplifies `calwebb_spec2` and keeps the internal workings of `imprint_step` contained.

Agreed that this does add some rerunning of code, since that determination will be done for every invocation of `imprint_step`. However, it is a low-cost rerun and the aforementioned simplification of the logic is worth it, IMHO.

Reply:
The comment for this line of code says that you're finding the dither positions of the leakcal exposures, but `members_by_type` contains the list of *all* ASN members. Or is the `imprint_pos_no` function limiting the search to only leakcal members?